compiler/extensions/cpp/runtime/src/zserio/BitStreamReader.cpp

```diff
@@ -201,11 +201,17 @@ namespace
         {
             checkEof(ctx, numBits);
 
-            ctx.cacheNumBits = static_cast<uint8_t>(ctx.bufferBitSize - ctx.bitIndex);
+            auto alignedBitSize = ctx.bufferBitSize - ctx.bitIndex + 7u;
+            alignedBitSize -= alignedBitSize % 8u;
+
+            ctx.cacheNumBits = static_cast<uint8_t>(alignedBitSize);
```

Comment:
Oops, this might be wrong.

Reply:
Note that in this version ctx.bufferBitSize is always rounded to full bytes and ctx.bitIndex is only moving by multiple of 8, so it's also always rounded to full bytes.